test/integration/consul-container/test/upgrade/healthcheck_test.go

```diff
@@ -256,7 +256,15 @@ func clientsCreate(t *testing.T, numClients int, image string, version string, c
 }
 
 func serviceCreate(t *testing.T, client *api.Client, serviceName string) uint64 {
-	err := client.Agent().ServiceRegister(&api.AgentServiceRegistration{Name: serviceName, Port: 9999})
+	err := client.Agent().ServiceRegister(&api.AgentServiceRegistration{
+		Name: serviceName,
+		Port: 9999,
+		Connect: &api.AgentServiceConnect{
+			SidecarService: &api.AgentServiceRegistration{
+				Port: 22005,
+			},
```

Comment:
No objection to adding this but is there something we wanted to validate by adding it?

Reply:
@DanStough , thanks for your comments. This is added to break the upgrade from 1.12.x to 1.13.0. In the future, we plan to add more struct to the snapshot to avoid any upgrade failure.

Reply:
link to the reported bug: https://github.com/hashicorp/consul/issues/14107